server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -183,8 +205,16 @@ public void close() {
     }
 
     void clear(CacheEntity entity) {
-        keysToClean.add(new CleanupKey(entity, null));
+        CleanupKey cleanupKey = new CleanupKey(entity, null);
+        keysToClean.put(cleanupKey, new CleanupStatus());
+        updateStaleKeysInDiskCount(cleanupKey);
         cleanCache();
+        /*
+        this would be triggered by the cache clear API call
+        we need to make sure we clean the disk cache as well
+        hence passing threshold as 0
+        */
+        cleanDiskCache(0);
```

Comment:
I'll wait addressing this until your framework changes are in, it will give me a better picture on how to move all of this into each cache layer. Ideally, yes i also want the cleanup logic of each cache to be in their own classes instead of indicesrequestcache.java, which will also address this.